data/awsrds/v0/main.go

```diff
@@ -0,0 +1,448 @@
+	var db *sqlx.DB
+	for _, engine := range engines {
+		dsn := fmt.Sprintf(engine+"?tls=true&tls=skip-verify&allowCleartextPasswords=true",
+			cfg.DBUser, authToken, DBEndpoint, cfg.DBName,
+		)
+
+		db, err = sqlx.Open("mysql", dsn)
+		if err == nil {
+			return db, nil
+		}
+	}
+
+	return nil, fmt.Errorf("error creating SQL client")
+}
+
+func getUser(setup *structpb.Struct) string {
+	return setup.GetFields()["user"].GetStringValue()
+}
+func getPassword(setup *structpb.Struct) string {
+	return setup.GetFields()["password"].GetStringValue()
+}
+func getName(setup *structpb.Struct) string {
+	return setup.GetFields()["name"].GetStringValue()
+}
+func getHost(setup *structpb.Struct) string {
+	return setup.GetFields()["host"].GetStringValue()
+}
+func getPort(setup *structpb.Struct) string {
+	port := setup.GetFields()["port"].GetNumberValue()
+	portStr := strconv.FormatFloat(port, 'f', -1, 64)
+	return portStr
+}
+func getRegion(setup *structpb.Struct) string {
+	return setup.GetFields()["region"].GetStringValue()
+}
+func getAWSAccessKeyID(setup *structpb.Struct) string {
+	return setup.GetFields()["aws-access-key-id"].GetStringValue()
+}
+func getAWSSecretAccessKey(setup *structpb.Struct) string {
+	return setup.GetFields()["aws-secret-access-key"].GetStringValue()
+}
+
+func (e *execution) Execute(ctx context.Context, inputs []*structpb.Struct) ([]*structpb.Struct, error) {
+	outputs := []*structpb.Struct{}
+
+	var client *sqlx.DB
+	var err error
+
+	// check if the password field is present, if it is, create a client with username and password else create a client with IAM
+	if e.Setup.GetFields()["password"] != nil {
+		client, err = newClient(e)
+	} else if e.Setup.GetFields()["password"] == nil && e.Setup.GetFields()["aws-access-key-id"] != nil && e.Setup.GetFields()["aws-secret-access-key"] != nil {
+		client, err = newClientIAM(e)
+	} else {
+		return nil, fmt.Errorf("some fields are missing, please fill in the required fields")
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("error creating SQL client: %v", err)
+	}
+	defer client.Close()
+
+	for _, input := range inputs {
+		var output *structpb.Struct
+
+		switch e.Task {
+		case taskUpsert:
+			tableName := input.GetFields()["table-name"].GetStringValue()
+
+			var err error
+			if ListValue, ok := input.GetFields()["data"].GetKind().(*structpb.Value_ListValue); ok {
+				for _, data := range ListValue.ListValue.Values {
+					err = UpsertUser(client, &tableName, data.GetStructValue())
+					if err != nil {
+						return nil, err
+					}
+				}
+			} else {
+				// Unmarshal the JSON data from the protobuf into structpb.Struct
+				data := input.GetFields()["data"].GetStructValue()
+
+				err = UpsertUser(client, &tableName, data)
+			}
+
+			if err != nil {
+				return nil, err
+			}
+
+			output = &structpb.Struct{Fields: map[string]*structpb.Value{
+				"status": {Kind: &structpb.Value_StringValue{StringValue: "Successfully executed SQL statement"}}}}
```

Comment:
```
output = &structpb.Struct{Fields: map[string]*structpb.Value{
				"status": {Kind: &structpb.Value_StringValue{StringValue: "Successfully executed SQL statement"}}}}
```

Same as comment above.
If you use the component struct, you can have the benefit of making code cleaner.

[Example PR](https://github.com/instill-ai/component/pull/190/files#diff-e489ebc6b09b1110ffef7a388d25d6199476e1ec1cfc19c60db73e39735bd98dL205)